### ISSUE TITLE
[java] AvoidBranchingStatementAsLastInLoop: False-negative about try/finally block

### ISSUE THREAD

**Issue by LynnBroe** [pmd/pmd#4063]
**Affects PMD Version:** 

**Rule:** [AvoidBranchingStatementAsLastInLoop](https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#avoidbranchingstatementaslastinloop)

**Description:**

**Code A: PMD report  Avoidbranchingstatementaslastinloop error at line 11**

```java
while (true) {
  try {
    // nothing to do
  }
  catch (Throwable c) {
    throw c;
  }
  break; // violation here
}
```
**Code B: transform the code,then the PMD doesn't report any error**

```java
while (true) {
  try {
    // nothing to do
  }
  catch (Throwable c) {
    throw c;
  }
  finally {
    break; // missing violation
  }
}
```

**Running PMD through:** *CLI*
